tests/Unit/UseCases/CreatePayment/CreatePaymentUseCaseTest.php

```diff
@@ -240,6 +270,37 @@ public function testCreateDirectDebitPaymentWithInvalidIbanFails(): void {
 		$this->assertEquals( "An invalid Iban was provided", $result->errorMessage );
 	}
 
+	public function testPaymentResponseContainsRequestedURLGenerator(): void {
+
+		$urlGeneratorMock = $this->createMock(CreditCard::class );
+
+		$urlFactoryMock = $this->createMock(PaymentURLFactory::class);
+		$urlFactoryMock->method('createURLGenerator')->willReturn( $urlGeneratorMock );
+
+		$useCase = new CreatePaymentUseCase(
+			$this->makeFixedIdGenerator(),
+			$this->makePaymentRepository(
+				new CreditCardPayment(
+					self::PAYMENT_ID,
+					Euro::newFromCents( 100 ),
+					PaymentInterval::OneTime
+				)
+			),
+			$this->makePaymentReferenceGeneratorStub(),
+			$this->makeValidateIbanUseCase(),
+			$this->makePaymentProviderURLFactoryDummy()
```

Comment:
This test is about using the real factory, so you can't use the dummy here (which is the reason this test fails).

I'd recommend to mock the factory a bit more: It should `expect` an `AdditionalPaymentData` with all data from the request (for full "coverage" you could do a sofort request that contains the payment reference code) and it `willReturn` a dummy payment URL generator. 

Later in the test you can assert that the URL generator returned in the success result is the same dummy URL generator you returned from the factory